io/input/chunk_inputformat.cpp

```diff
@@ -45,20 +79,24 @@ bool ChunkInputFormat::next(boost::string_ref& ref) {
     if (buffer_.size() == 0) {
         // fetch new block
         bool success = fetch_new_block();
-        if (!success)
+        if (!success) {
+            is_setup_ = ChunkInputFormatSetUp::NotSetUp;
             return false;
+        }
     }
     if (r == buffer_.size()) {
         // last character in block
         bool success = fetch_new_block();
-        if (!success)
+        if (!success) {
+            is_setup_ = ChunkInputFormatSetUp::NotSetUp;

```

Comment:
I see, I will make changes. But what about the case where the user first do `infmt.set_input(a);` , reads some part of the files and then decides to read from another file `infmt.set_input(b);`. Is this allowed, i.e. we just discard the previous file? Or we should forbid such behaviour and throw an exception?